lib/kcl.js

```diff
@@ -153,41 +154,70 @@ module.exports = function(config, kinesis) {
       return delete instanceShardList[shard.id];
     }
 
-    // expose the checkpoint function to the procerssRecords, so they can checkpoint
+    // expose the checkpoint function to the processRecords, so they can checkpoint
     // without requesting more records.
     shard.checkpointFunc = db.checkpoint.bind(null, shard);
 
-    kinesis.getRecords(
-      { ShardIterator: shard.iterator, Limit: config.limit },
-      function(err, resp) {
-        if (err && err.code === 'SyntaxError') return setTimeout(getRecords.bind(this, shard), 100);
-        if (err) throw err;
+    if (config.verbose) logger.info('[ getRecords ] params:', JSON.stringify({
+      ShardIterator: shard.iterator,
+      Limit: config.limit
+    }));
 
-        if (config.cloudwatchNamespace &&
-          resp.Records &&
-          resp.Records[0] &&
-          resp.Records[0].ApproximateArrivalTimestamp) {
-          throttledPutToCloudwatch('ShardIteratorAgeInMs', (+new Date()) - resp.Records[0].ApproximateArrivalTimestamp, 'Milliseconds', shard.id);
-        }
+    var req = kinesis.getRecords({ShardIterator: shard.iterator, Limit: config.limit});
+    var requestTimeOut = setTimeout(req.abort.bind(req), 5000);
```

Comment:
Why not handle this through the `httpOptions` [timeout and connectTimeout](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#constructor-property) and avoid the `setTimeout` that you have to manage yourself?